pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java

```diff
@@ -123,21 +139,27 @@ protected void startController(ControllerConf config) {
 
     String zkStr = config.getZkStr();
     _zkClient = new ZkClient(zkStr);
-    if (_zkClient.exists("/" + helixClusterName)) {
+    if (_zkClient.exists("/" + helixClusterName) && deleteCluster) {
       _zkClient.deleteRecursive("/" + helixClusterName);
     }
 
     startControllerStarter(config);
 
-    _helixManager = _helixResourceManager.getHelixZkManager();
-    _helixAdmin = _helixResourceManager.getHelixAdmin();
-    _propertyStore = _helixResourceManager.getPropertyStore();
+    // HelixResourceManager is null in Helix only mode, while HelixManager is null in Pinot only mode.
+    if (_helixManager != null) {
```

Comment:
should check mode here rather than helixManager being null. You can assert it being null or not so depending on the mode

Reply:
Done.